### PR TITLE
knife-windows WinRM Listener Create should default to blank hostname if not specified or allow --nohostname to be specified.

### PR DESCRIPTION
**Pull request by prabhu-das**
Updated hostname default option to blank for knife windows listner create and removed * as default hostname

### CODE REVIEW

File: lib/chef/knife/windows_listener_create.rb [hunk 1]
```diff
@@ -40,7 +40,7 @@ class WindowsListenerCreate < Knife
         :short => "-h HOSTNAME",
         :long => "--hostname HOSTNAME",
         :description => "Hostname on the listener. Default is *",
```
Comment (NimishaS) [inline lib/chef/knife/windows_listener_create.rb]: Description should be changed as well.

**Comment by NimishaS** [comment]
Default value of hostname(common name) is "*" while certificate generation. Should it be consistent with this behavior?

**Comment by adamedx** [comment]
I think * is not appropriate in either case.

**Comment by btm** [comment]
The hostname used in cert generate goes into the CN, right? I would expect it has to be something and we should raise an error if it isn't provided. I'm presuming winrm is going to be matching the hostname against the CN and give us an error if it doesn't match.

Have we tested connecting to a system with a certificate with a CN of * or blank?

**Comment by NimishaS** [comment]
@btm , yes hostname in cert generate goes to CN. @adamedx had suggested to set it to local fqdn when not provided.

We tested that connecting to a system works doesn't work when CN is blank.

**Comment by adamedx** [comment]
Correct, we'll assume a "local" workflow where you are trying to set up the listener for the system you're running on. In that case, you want the cert to be the fqdn (assuming fqdn is also the server name in the listener url, which should be common).

Otherwise, you need to set this yourself to match the url of the listener.

**Comment by btm** [comment]
Taking https://github.com/chef/knife-windows/pull/201 into consideration, we should mention CN because that's meaningful to a lot of people who know something or are reading something about certificates.

```
:description => "Use to specify the remote hostname for the CN field of the certificate.
    For example, --hostname windows_node.example.org or *.example.org
    Default value is the hostname of the local system."
```

**Comment by NimishaS** [comment]
@btm, we'll fix the description in a separate PR where we set default hostname of certificate to local fqdn.

**Comment by NimishaS** [comment]
Something got wrong during rebase. Created https://github.com/chef/knife-windows/pull/205 cherry-picking the changes from this PR.

**Pull request closed without merging**